src/freehand_selection.jl

```diff
@@ -1,61 +1,188 @@
 function init_freehand_select(ctx::ImageContext)
     c = ctx.canvas
     enabled = Signal(true)
-    dragging = Signal(false)
+    drawing = Signal(false)
+    moving = Signal(false)
+    diff = Signal(XY(NaN,NaN))
 
     dummybtn = MouseButton{UserUnit}()
     
     sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
         if !ispolygon(value(ctx.points)) || !isinside(Point(btn.position), Point.(value(ctx.points))) # prevents conflict with init_move_polygon
-            push!(dragging, true)
+            push!(drawing, true)
             push!(ctx.shape, Rectangle()) # some identifier of type of selection
             push!(ctx.points, [])
```

Comment:
Since a `Polygon` might have quite a few vertices, perhaps here it makes some sense to be a bit performance-sensitive. Here's some [reading](https://docs.julialang.org/en/latest/manual/performance-tips/#Avoid-containers-with-abstract-type-parameters-1), particularly "Avoid containers with abstract type parameters" and all subsections under "Type declarations" (particularly the first two). This is creating a `Vector{Any}`, and it might be better if it were a `Vector{XY{Float64}}` or something.